addon/PreviewProvider.js

```diff
@@ -219,6 +219,9 @@ PreviewProvider.prototype = {
       if (existingLinks.has(link.cache_key)) {
         return Object.assign({}, existingLinks.get(link.cache_key), link);
       }
+      if (!link.metadata_source) {
+        link.metadata_source = "TippyTopProvider";
```

Comment:
Can we move this to right here:

https://github.com/mozilla/activity-stream/blob/master/addon/TippyTopProvider.js#L48